Various test cleanups/fixes

martinpitt opened the pull request:
- Adjust documentation for running the integration tests
 - Clean up checks for "am I running on an Atomic image"

martinpitt commented:
Ack, force-pushed without the pyflakes changes.

martinpitt commented:
I pushed another commit to fix `TestKeys.testAuthorizedKeys` not always getting stuck in a root shell.

dperpeet commented on test/common/testvm.py:
```diff
@@ -107,6 +107,7 @@ def __init__(self, address=None, image=None, verbose=False, label=None, fetch=Tr
         self.arch = "x86_64"
 
         self.image = image or testinfra.DEFAULT_IMAGE
+        self.atomic_image = self.image in ["rhel-atomic", "fedora-atomic", "continuous-atomic"]
```
What is the reason for this change? It barely saves any code in the tests and adds an extra place where we need maintenance when the list of images changes.

martinpitt replied:
This list was repeated in the existing code several times, so this centralizes that list. Other places used `"atomic" in m.image` which might lead to unintended test skips, and the tests should be consistent in how they check for "am I running on a known atomic image".

stefwalter replied:
The problem is that many of the cases where ```m.atomic.image``` is used in this pull request should not be consistent in how they check this. The various Atomic's gain features at different rates.

dperpeet replied:
maybe we can add the list of atomics in testinfra, then we'd have them all in the same place: https://github.com/cockpit-project/cockpit/blob/master/test/common/testinfra.py#L53

martinpitt replied:
@dperpeet: indeed, that's a nicer place; I originally was thinking to add a new flag ("is Atomic") to the `DEFAULT_VERIFY` map entries, but that smells like over-engineering. Simply moving the list into that file should do fine IMHO.

stefwalter commented on test/verify/check-docker-storage:
```diff
@@ -38,7 +38,7 @@ import unittest
 def can_manage(machine, vg):
     # Atomic is all set.
-    if machine.image in [ "rhel-atomic", "fedora-atomic", "continuous-atomic" ]:
+    if machine.atomic_image:
```
This has not always been the case for all Atomics. I would suggest leave the original check in place, so we can add/remove them individually.

stefwalter commented on test/verify/check-pages:
```diff
@@ -48,7 +48,7 @@ WantedBy=default.target
         # On Atomic no locales other than en_US are supported on the host itself
         # HACK: https://bugzilla.redhat.com/show_bug.cgi?id=1186757
-        elif m.image in [ "continuous-atomic", "fedora-atomic", "rhel-atomic" ]:
+        elif m.atomic_images:
```
This won't always be the case for all atomics. There's a bug fix in progress, and it'll likely be released on these one by one.

martinpitt replied:
Typo here, needs to be "image". (I'll wait for the other test results to finish before I re-push).

martinpitt commented:
Updated again, I moved the list into `testinfra.py` (but still kept the `.atomic_image` flag so that tests don't need to import `testinfra` just for this) and kept the explicit lists for the `HACK`s.

stefwalter commented:
Looks like this needs a rebase now that #5599 was merged.

martinpitt commented:
Rebased, conflict fixed, force-pushed.

martinpitt commented:
Force-pushed again with fixing the three new pyflakes errors in check-os-tree, they landed in master before landing the pyflakes branch.

martinpitt commented:
The ubuntu-1604 test failure looks like a race condition? I tried to restart it with
```
 test/github-trigger 5701 verify/ubuntu-1604
verify/ubuntu-1604: triggering on pull request 5701
statuses/f83ebef90178aa460130bc2a10376bb05bf9d2ed
{"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
Traceback (most recent call last):
  File "test/github-trigger", line 95, in <module>
    sys.exit(main())
  File "test/github-trigger", line 89, in main
    return trigger_pull(github, opts)
  File "test/github-trigger", line 69, in trigger_pull
    github.post("statuses/" + revision, changes)
  File "/home/martin/upstream/cockpit/test/common/testinfra.py", line 420, in post
    raise Exception("GitHub API problem: {0}".format(response['reason'] or status))
Exception: GitHub API problem: Not Found
```

I'm not sure what this is trying to say -- https://api.github.com/repos/cockpit-project/cockpit/statuses/f83ebef90178aa460130bc2a10376bb05bf9d2ed does exist and looks valid. But I only used this script to trigger image builds so far, not for re-triggering tests.

petervo commented:
Probably a permissions issue. Triggered it for you.
